### PR TITLE
Add `pydoc` enforcement via `ruff`

### PR DESCRIPTION
**Pull request by michplunkett**
## Describe your changes
Added `pydoc` enforcement mechanism to ruff via the `--config` flag as well as a `Makefile` to simplify commands.

## Checklist before requesting a review
- [x] pre-commit run --all-files (run before pushing)
- [x] pytest if applicable
- [x] Update relevant documentation if applicable: doc strings, readme, poetry.

### CODE REVIEW

**Review by FedericoDM — COMMENTED** [review]
No ruff, thanks >;(

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -25,15 +25,16 @@ repos:
...
-    rev: v0.0.286
+    rev: v0.0.289
     hooks:
       - id: ruff
         types_or: [python, pyi, jupyter]
         args:
           - --fix
           - --target-version=py311
-          - --select=B,C,E,F,W
+          - --select=B,C,E,F,W,D
```
Comment (michplunkett) [inline .pre-commit-config.yaml:35]: The `D` option makes it so ruff looks at documentation: https://docs.astral.sh/ruff/rules/#pydocstyle-d

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -25,15 +25,16 @@ repos:
...
           - --line-length=80
+          - --config=pyproject.toml
```
Comment (michplunkett) [inline .pre-commit-config.yaml:37]: Designates the configuration file.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Here is some general information on Makefile's so that you can grow this out:
+# https://www.gnu.org/software/make/manual/html_node/Introduction.html
+default: lint
```
Comment (michplunkett) [inline Makefile:3]: You have yourself a `Makefile`, @FedericoDM.

**Review by FedericoDM — DISMISSED** [review]

File: pyproject.toml [hunk 4]
```diff
@@ -20,6 +20,11 @@ flask-sqlalchemy = "^3.1.1"
 faker = "^25.0.0"
 flask-login = "^0.6.3"
 
+[tool.ruff.pydocstyle]
+convention = "google"
```
Comment (michplunkett) [inline pyproject.toml:24]: Add requirement for `google` style documentation: https://docs.astral.sh/ruff/faq/#does-ruff-support-numpy-or-google-style-docstrings

File: pyproject.toml [hunk 5]
```diff
@@ -20,6 +20,11 @@ flask-sqlalchemy = "^3.1.1"
...
+
+[tool.ruff.per-file-ignores]
+"new_arrivals_chi/migrations/*" = ["D103"]
```
Comment (michplunkett) [inline pyproject.toml:27]: Removes the `D103` rule from applying to the `new_arrivals_chi/migrations/*` folder.

**Comment by michplunkett** [comment]
Someone needs to make the changes before this is good to go, but it's otherwise ready to do some documentation enforcement.

**Review by FedericoDM — APPROVED** [review]

File: new_arrivals_chi/app/templates/info.html [hunk 6]
```diff
@@ -20,15 +20,15 @@
 {% block content %}
 
 <form action="/info" method="GET">
-    <button type="submit" class="language-button">{{ 'Español' if language == 'en' else 'English' }}</button>
+    <button type="submit" class="language-button">{{ 'Español' if language == 'es' else 'English' }}</button>
```
Comment (michplunkett) [inline new_arrivals_chi/app/templates/info.html:23]: A simpler way of doing this going forward may be to send all of the text for a given file in some object with the HTTP response. That way you can just use the language selected in in the app overall, and then not have to worry about it in the templates.

Comment (FedericoDM) [reply new_arrivals_chi/app/templates/info.html:23]: @michplunkett I am currently working on substituting this with some translation JSONs in this branch:

https://github.com/uchicago-capp-30320/new-arrivals-chi/tree/add-json-languages

Does that make sense to you?

**Review by FedericoDM — APPROVED** [review]

**Review by MadeleineKRoberts — APPROVED** [review]

**Pull request merged**